pkg/gitrepo/repo.go

```diff
@@ -303,6 +310,46 @@ func (r *Repo) ResolveVersion(ctx context.Context, ref string) (string, error) {
 	return pseudoVersion, nil
 }
 
+// GetFileContent retrieves content of file stored at path on version specified in options.
+func (r *Repo) GetFileContent(path string, options CheckoutOptions) ([]byte, error) {
+	repo, err := git.Open(r.storage, r.worktree)
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	worktree, err := repo.Worktree()
+	if err != nil {
+		return nil, microerror.Mask(err)
+	}
+
+	// When empty CheckoutOptions defaults to master branch.
+	opt := &git.CheckoutOptions{}
+	if options.Branch != "" {
+		opt.Branch = plumbing.NewRemoteReferenceName("origin", options.Branch)
```

Comment:
I don't like adding `origin` implicitly there. We do similar thing in `ResolveVersion` so I think it would be good to have this aligned. How about replacing `options CheckoutOptions` with `ref string` and reusing code from `ResolveVersion`?

Reply:
Do you mean using this ?

```
		hash, err := repo.ResolveRevision(plumbing.Revision(ref))
```

Reply:
done